MdeModulePkg/Bus/Pci/SdMmcPciHcDxe/SdMmcPciHcDxe.c

```diff
@@ -647,7 +647,7 @@ SdMmcPciHcDriverBindingStart (
   }
 
   Support64BitDma = TRUE;
-  for (Slot = FirstBar; Slot < (FirstBar + SlotNum); Slot++) {
+  for (Slot = FirstBar; Slot < (UINT8)(FirstBar + SlotNum); Slot++) {

```

Comment:
Why was this an issue? The comparison is for UINT8 < (UINT8 + UINT8), shouldn't that not need casting? What am I missing?

Reply:
Same question below.

Reply:
C has the ability to promote arithmetic operations in expressions to larger types to cover the potential resulting precision. UINT8 + UINT8 could be larger than UINT8.

I think it is fair and technically correct to say the type from the addition is larger than a UINT8.

The "correct" resolution would be to perform the addition with SafeIntLib, which explicitly checks for overflow, and ensure the result is a UINT8 for the comparison. I mentioned offline the other day, this is everywhere in the codebase so I didn't go to that length here.

However, I think I'll go down that path since it ensures functional correctness regardless of the unlikelihood of overflow in this particular case. I'll make that update soon.

Reply:
Updated to use SafeIntLib in PR commit https://github.com/microsoft/mu_basecore/pull/257/commits/7032915c38b3dfc09b8d9a9a774379f9c1cabb06